### PR TITLE
Rename / Rearrange BinaryCache functions

### PR DESCRIPTION
**Pull request by ras0219-msft**
This PR extracts ancillary changes from #998 so they can be reviewed separately from the primary rewrite of binarycaching.cpp.

- `create_binary_providers_from_configs_pure` -> `parse_binary_provider_configs`
- Different callers of `Install::perform` performed different amounts of "prep" (computing ABIs, etc). This work was then repeated inside `perform`.

    To resolve this, all callers are now expected to compute all ABIs ahead of time. I've removed the `var_provider` parameter, and made the `action` parameter `const`.
- I've lifted the `REQUIRE_LINES` test macro so it can be used in more places.

### CODE REVIEW

File: src/vcpkg/commands.upgrade.cpp [hunk 1]
```diff
@@ -206,8 +205,11 @@ namespace vcpkg::Commands::Upgrade
 
         var_provider.load_tag_vars(action_plan, provider, host_triplet);
 
-        const InstallSummary summary = Install::perform(
-            args, action_plan, keep_going, paths, status_db, binary_cache, null_build_logs_recorder(), var_provider);
+        BinaryCache binary_cache(args, paths);
+        binary_cache.prefetch(action_plan.install_actions);
+        compute_all_abis(paths, action_plan, var_provider, status_db);
```
Comment (autoantwort) [inline src/vcpkg/commands.upgrade.cpp:210]: Are the abis not needed for the prefetch?

Comment (ras0219-msft) [reply src/vcpkg/commands.upgrade.cpp:210]: Nice catch! This is entirely my error.

**Review by BillyONeal — APPROVED** [review]

File: src/vcpkg/commands.install.cpp [hunk 2]
```diff
@@ -552,8 +551,6 @@ namespace vcpkg
                 perform_install_plan_action(args, paths, action, status_db, binary_cache, build_logs_recorder));
         }
 
-        compute_all_abis(paths, action_plan, var_provider, status_db);
-        binary_cache.prefetch(action_plan.install_actions);
```
Comment (autoantwort) [inline src/vcpkg/commands.install.cpp:556]: Now the `prefetch` is done before `Install::execute_plan`, but in line 544/543 the packages folder is removed.

Comment (BillyONeal) [reply src/vcpkg/commands.install.cpp:556]: Those lines are only relevant for packages that are being removed.

Comment (autoantwort) [reply src/vcpkg/commands.install.cpp:556]: But if a package gets reinstalled with a different feature set it gets removed and then installed

**Pull request merged**